src/mappings/helpers.ts

```diff
@@ -592,32 +592,28 @@ function max(a: BigDecimal, b: BigDecimal): BigDecimal {
 }
 
 export function calculateOwnStakeRatio(indexer: Indexer): BigDecimal {
+  let stakedTokensBD = indexer.stakedTokens.toBigDecimal()
+  let delegatedTokensBD = indexer.delegatedTokens.toBigDecimal()
   let graphNetwork = GraphNetwork.load('1')
-  let delegationRatio = BigInt.fromI32(graphNetwork.delegationRatio)
-  let maxPossibleTotalUsable =
-    indexer.stakedTokens.toBigDecimal() +
-    (indexer.stakedTokens.toBigDecimal() * delegationRatio.toBigDecimal())
-  let currentTotalStake =
-    indexer.stakedTokens.toBigDecimal() + indexer.delegatedTokens.toBigDecimal()
+  let delegationRatioBD = BigInt.fromI32(graphNetwork.delegationRatio).toBigDecimal()
+  let maxPossibleTotalUsable = stakedTokensBD + stakedTokensBD * delegationRatioBD
+  let currentTotalStake = stakedTokensBD + delegatedTokensBD
   let totalUsable = min(maxPossibleTotalUsable, currentTotalStake)
   return totalUsable == BigDecimal.fromString('0')
     ? BigDecimal.fromString('0')
-    : indexer.stakedTokens.toBigDecimal() / totalUsable
+    : stakedTokensBD / totalUsable
 }
 
 export function calculateDelegatedStakeRatio(indexer: Indexer): BigDecimal {
+  let stakedTokensBD = indexer.stakedTokens.toBigDecimal()
+  let delegatedTokensBD = indexer.delegatedTokens.toBigDecimal()
   let graphNetwork = GraphNetwork.load('1')
-  let delegationRatio = BigInt.fromI32(graphNetwork.delegationRatio)
-  let maxPossibleTotalUsable =
-    indexer.stakedTokens.toBigDecimal() +
-    (indexer.stakedTokens.toBigDecimal() * delegationRatio.toBigDecimal())
-  let totalUsable = min(
-    maxPossibleTotalUsable,
-    indexer.stakedTokens.toBigDecimal() + indexer.delegatedTokens.toBigDecimal(),
-  )
+  let delegationRatioBD = BigInt.fromI32(graphNetwork.delegationRatio).toBigDecimal()
+  let maxPossibleTotalUsable = stakedTokensBD + stakedTokensBD * delegationRatioBD
+  let totalUsable = min(maxPossibleTotalUsable, stakedTokensBD + delegatedTokensBD)
   return totalUsable == BigDecimal.fromString('0')
     ? BigDecimal.fromString('0')
-    : indexer.delegatedTokens.toBigDecimal() / totalUsable
+    : delegatedTokensBD / totalUsable
```

Comment:
I am seeing a weird number here for overdelegated scenarions. See for 0xa3276e7ab0a162f6a3b5aa6b3089accbaa65d12e . In the subgraph we have about 99.5% right now but in the google sheet around 94%. 

The formulas are just different, from what I can tell. Maybe I am missing something though. Let me know if this is a bug , or if this was meant to be

Reply:
It should be now working as expected on the dev-subgraph deploy

Reply:
looks good!